src/storage.ts

```diff
@@ -58,6 +58,13 @@ export interface RetryOptions {
   retryableErrorFn?: (err: ApiError) => boolean;
 }
 
+export interface PreconditionOptions {
+  ifGenerationMatch?: number;
+  ifGenerationNotMatch?: number;
+  ifMetagenerationMatch?: number;
+  ifMetagenerationNotMatch?: number;
```

Comment:
Is there any ETag support in this library?

Reply:
@tritone not currently. I opened a buganizer issue to do the implementation of ETag support.

Reply:
Sounds good!